config/jobs/image-pushing/k8s-staging-infra-tools.yaml

```diff
@@ -47,85 +47,120 @@ postsubmits:
-              - --env-passthrough=PULL_BASE_REF
-              - dns/octodns-docker
-    - name: post-k8sio-push-image-k8s-infra
-      cluster: k8s-infra-prow-build-trusted
-      annotations:
-        testgrid-dashboards: sig-k8s-infra-k8sio, sig-k8s-infra-gcb
-      decorate: true
-      run_if_changed: "^images/k8s-infra/"
-      branches:
-        - ^main$
-      spec:
-        serviceAccountName: gcb-builder
-        containers:
-          - image: gcr.io/k8s-staging-test-infra/image-builder:v20230111-cd1b3caf9c
-            command:
-              - /run.sh
-            args:
-              - --project=k8s-staging-infra-tools
-              - --scratch-bucket=gs://k8s-staging-infra-tools-gcb
-              - --env-passthrough=PULL_BASE_REF
-              - images/k8s-infra
-    - name: post-k8sio-push-image-public-log-asn-matcher
-      cluster: k8s-infra-prow-build-trusted
-      annotations:
-        testgrid-dashboards: sig-k8s-infra-k8sio, sig-k8s-infra-gcb
-        testgrid-alert-email: public-log-asn-matcher@ii.coop
-        description: Builds the public-log-asn-matcher image for generating data that is used for reviewing Kubernetes Public artifact traffic in DataStudio.
-      decorate: true
-      run_if_changed: "^images/public-log-asn-matcher/"
-      branches:
-        - ^main$
-      spec:
-        serviceAccountName: gcb-builder
-        containers:
-          - image: gcr.io/k8s-staging-test-infra/image-builder:v20230111-cd1b3caf9c
-            command:
-              - /run.sh
-            args:
-              # this is the project GCB will run in, which is the same as the GCR images are pushed to.
-              - --project=k8s-staging-infra-tools
-              - --scratch-bucket=gs://k8s-staging-infra-tools-gcb
-              - --env-passthrough=PULL_BASE_REF
-              - images/public-log-asn-matcher
-    - name: post-k8sio-push-image-cs-fetch-repos-canary
-      cluster: k8s-infra-prow-build-trusted
-      annotations:
-        testgrid-dashboards: sig-k8s-infra-k8sio, sig-k8s-infra-gcb
-        description: Builds the cs-fetch-repos image which creates a git repository configuration file for indexing by codesearch (aka Hound).
-      decorate: true
-      run_if_changed: "^images/codesearch/cs-fetch-repos/"
-      branches:
-        - ^main$
-      spec:
-        serviceAccountName: gcb-builder
-        containers:
-          - image: gcr.io/k8s-staging-test-infra/image-builder:v20230111-cd1b3caf9c
-            command:
-              - /run.sh
-            args:
-              - --project=k8s-staging-infra-tools
-              - --scratch-bucket=gs://k8s-staging-infra-tools-gcb
-              - --env-passthrough=PULL_BASE_REF
-              - images/codesearch/cs-fetch-repos
+  # Special job to deploy the staging registry
+  # this should closely match post-registry-push-images
+  # TODO(bentheelder): rerun_auth_config (let a github team re-run these, I probably need to join a team...)
```

Comment:
+1

Reply:
As a fallback I can currently do this via the prow-wide rerun team. I think that also applies to the rest of test-infra-oncall and the K8s Infra Leads, which is a pretty good starting point for who can re-run these.

I think it would be entirely appropriate to have a k8s-infra specific group with rerun permission for this whole file though.